builder/dockerfile/internals.go

```diff
@@ -248,17 +247,12 @@ func (b *Builder) download(srcURL string) (fi builder.FileInfo, err error) {
 	progressOutput := stdoutFormatter.StreamFormatter.NewProgressOutput(stdoutFormatter.Writer, true)
 	progressReader := progress.NewProgressReader(resp.Body, progressOutput, resp.ContentLength, "", "Downloading")
 	// Download and dump result to tmp file
+	// TODO: add filehash directly
```

Comment:
Because this is a single file there isn't really a need for `LazyContext` and reading file back from the disk. We could just create a single hash object and pipe data through it. But that is an optimization for edge case so not currently implemented.